### PR TITLE
(#2522) Remove deprecated ScriptsCheckLastExitCode

### LINKED ISSUE #2522: Remove ScriptsCheckLastExitCode feature
As noted in these locations:
https://github.com/chocolatey/choco/blob/develop/src/chocolatey/infrastructure.app/builders/ConfigurationBuilder.cs#L311
https://github.com/chocolatey/choco/blob/develop/src/chocolatey/infrastructure.app/configuration/ChocolateyConfiguration.cs#L413-L414

The ScriptsCheckLastExitCode feature was slated to be removed in v0.11.0. Since we are now at v0.11.3, that did not happen. Should the removal be retargeted for 1.0.0?

### PR DESCRIPTION
**Pull request by gep13**
## Description Of Changes

The ScriptsCheckLastExitCode feature, has been removed.  This 
was marked as going to be removed in 0.11.0 of Chocolatey,
however, there wasn't an issue created to capture this, and as a
result, this was missed.

## Motivation and Context

This was only intended to be a temporary feature while folks moved over
to using the recommended way to handle their packages.

## Testing

None.  This was a removal of functionality, and therefore there is no need
for any testing.

## Change Types Made

* [x] Removal of legacy code
* [ ] Bug fix (non-breaking change)
* [ ] Feature / Enhancement (non-breaking change)
* [ ] Breaking change (fix or feature that could cause existing functionality to change)

## Related Issue

Fixes #2522

## Change Checklist

* [x] Requires a change to the documentation
* [ ] Documentation has been updated
* [ ] Tests to cover my changes, have been added
* [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by gep13** [comment]
The required documentation change has been captured in this issue: https://github.com/chocolatey/choco/issues/2391

**Review by TheCakeIsNaOH — CHANGES_REQUESTED** [review]
There are still references to `ChocolateyCheckLastExitCode` in ApplicationParameters and in chocolateyScriptRunner

Not sure if you are intending to do it in this PR or in a standalone PR, but there is documentation about `scriptsCheckLastExitCode` in:
- src\chocolatey\infrastructure.app\templates\ChocolateyReadMeTemplate.cs#L141
- GenerateDocs.ps1#L520

**Comment by gep13** [comment]
@TheCakeIsNaOH Thanks for catching that! I did a naive search of the codebase for the feature name, and not the associated environment variable.

I have fixed this up, and it is ready for another review.

File: src/chocolatey.resources/helpers/chocolateyScriptRunner.ps1 [hunk 1]
```diff
@@ -52,17 +52,13 @@ $lastExecutableExitCode = $LASTEXITCODE
...
-} 
+}
 
 if (-not $scriptSuccess) {
  Write-Debug "The script exited with a failure."
-} 
-
-$exitCode = 0
-if ($env:ChocolateyCheckLastExitCode -ne $null -and $env:ChocolateyCheckLastExitCode -eq 'true' -and $lastExecutableExitCode -ne $null -and $lastExecutableExitCode -ne '') {
-  $exitCode = $lastExecutableExitCode
```
Comment (corbob) [inline src/chocolatey.resources/helpers/chocolateyScriptRunner.ps1:63]: I am 85% sure the answer to this is that we don't have to worry, but just to be sure: Are there any changes in chocolatey.extension that will need to be made for this?

**Comment by corbob** [comment]
Aside from the comment regarding chocolatey.extension, looks good to me.

**Review by TheCakeIsNaOH — APPROVED** [review]

File: src/chocolatey.resources/helpers/chocolateyScriptRunner.ps1 (hunk 1 above)
Comment (gep13) [reply src/chocolatey.resources/helpers/chocolateyScriptRunner.ps1:63]: That is a great question! I will take a look at that to confirm. This wouldn't stop this PR from going in, but potentially something we need to update over there!

**Comment by gep13** [comment]
@TheCakeIsNaOH @corbob thanks for taking a look at this! Will get this merged in.

**Pull request merged**